### DISCUSSION TITLE
Many-to-Many and WHERE...IN

### DISCUSSION THREAD

**Discussion by robkallen** [tywalch/electrodb#139] [General]
Hi @tywalch, I have a couple of scenarios I'm not quite sure how to model in electro.  Please let me know if these are possible.

1. Is it possible to model many-to-many relationships using collections?  My exact situation is where I have a user that can be a member of multiple groups, and a group that can have multiple users as members.
2. Is there a way to do a filter similar to a SQL WHERE...IN command, like `SELECT * FROM users WHERE userID IN (1, 2, 3)`?

Once again, thanks for your help.

**Comment by tywalch** [comment]
Hey @robkallen,

1. Yes collections allow you to write a single request and get multiple entity types back. To reason about this, consider the PK composite to be the foreign key that relates both entities. All members of the collection must implement the same PK, so when querying you can supply the PK composite(s) and get back all values for each entity that implements to that collection. [Here is playground demonstrating a collection that brings together users and groups](https://electrodb.fun/?#code/PQKgBAsg9gJgpgGzARwK5wE4Es4GcA0YuccYGeqCALgUQBYCG5YA7llXWAGbZwB2MXGBDAAUKKwBbAA5QMVMAG8wAUT5V2AT0IBlTADcsAY1IBfbhiiSwAIkRwjVSzABGNgNzijUPrgVUGFwRSAF5bTShUDAB9AKC4aL4GSTgPcVAwAFViDFV1LWExb18FVBywML44FjyNKk0ACkVRMDBJWEQALiUW1rB+Os1umzLMG3xe1pzDE2HJTWiGaWlxybB9TFwsH2GARhte0wnWhionLBdUKjxu5r6wUYwASRhbtdb66Thhv2w+AHMDvcjmt-pZUNIXm97h9NF8fucAUC+iD7lwsBg-AA5ZLfHowsCfPE2X5YJFrVF9BAMbG46EwokIv6AinHPpwSQMLAIen3RlgADkpIBArZrVMhzZZPgAA8bvi+o9cLy+t4EMFHNs+HMOS5NnQsCsxX1pABrFX3bwyKBba7dADaNjBkUhMBsAF1jWicAhXrYzcjgV6iOaFQSrbJbXjHY8Xh7g610Yg-STTYGURTDqIjkpCYFgmBTABKTyiYp+MDOiEVMBVGpqQZNXrteA8sP9fL1YaPVb3abGYnzRbLXt9DaYrV7ZGU07nS7XZXtquui2EuF4oWI-4C1mg8HSHEpVf8zfMndByW9aVwOWLu59Zdk-6rtUajQ7WwpSR6zEGo3vMAzVXVoIxtdhoydfc409ADEx9FMAwAyk+1De9wysSNwIdGCCTg5MfjTJDM3FbNCEUOJgmLUtywUJZpBrOswD0DAZjgJoHhyQhl0LEt0mAQkoErOBaMrfcyAcOQYDABgBGk9UwCgLgwHYIRRGAfivx-IQGiVIsOKfaSwFfBx3z4UQ6IAOmMzUfFwXoLM0-VDSaZcoUFIdlwFYt7P+KAGiLeyOH4BoYFOBgKgAPnbdSwFCgJc0ebCuP3bDC0OEsgA)

2. This depends on how you structure your keys. If your Primary Index is only made up of a `userId` your options become more limited. [Here's a playground to demonstrate how you might accomplish this in different ways](https://electrodb.fun/?#code/PQKgBAsg9gJgpgGzARwK5wE4Es4GcA0YuccYGeqCALgUQBYCG5YA7llXWAGbZwB2MXGBDAAUKKwBbAA5QMVMAG8wAUT5V2AT0IBlTADcsAY1IBfbhiiSwAIkRwjVSzABGNgNzijUPrgVUGFwRSAF5bTShUDAB9AKC4aL4GSTgPcVFvXwVUYgwARjAwvjgWVXUtAApFUTAwSVhEAC4lGtqwfg0qTWabHMwbfFba3MMTHslNaIZpaQGhsH1MXCwfHrybVtNB2oYqJywXVCo8Zuq2sD6MAEkYU-naruk4Hr9sPgBzDfOt+a4sDD8ADlks8WucHponi99h8vm0fucEAwgSC7uCwI9QTZXlhYfMEW04JIGFgEGjwZjmgByHEfKnbeGbBm4+AADxOYLal3J52kAGseeDvDIoMtjs0ANq9XI3GwAXQZ6L+iFutn5cO+irauAFnPRwtkYtBEoV9zayoQquxfI1jO+m1EWyUGMCwTApgAlJ4Mj4-BdcgAmQpgYqlNSdTRVVr1eBkvXtcpdHqXObnEbGLETKYzVNtRYAlZ8NZwgm7faHY64QWXG6CiFQ2y0z74rV-AFUYEpOsYyFYpu2gm1JEorvx+t9mHNzXzIkkuNnCm96lN+n4pmtFlwdlV+OXHcL3m6g9CqyG9jG03o2oWq3qs2D7VHs21A2i8+S6WYWWXq83l42+811qUxHUIZQ4jdT1vUyP1LgAZmDUMygjKNahjJp4w6LRk1yXNhgMDNxkmaZZi1fNllWWx1nXHY9mwCsOWPGtVWPcdoTeKd4S1d5LFQaRazHHsG2xScB1bf4R1BVihInDixPmYcO1RQTKUbUSWxnYlSW7VSaUnVd7WA5kBC3Rj5j3bt+W7F9TzfcUwClHjIn4mB5S1c4-zVAD0QfYYnyvMBXyND9mLc59uBwS1-1tYCgPdUDnQgswvS8X1slyAAWRCSmQyoF3Q+cZ0TbpbBTLV0zGWwsxIvCFiWQtixosAy3oo4zPOZidKXNS5I084nL4gTpNUkTeunDyJKU0dhu60bcU4oyFORKapLNEb+z6wktMK9FdJXLUQMW2pN23atcks-yryC98HM-a5XJ-JVItvbzwV8ohLv1Wzgocx7wU860Yvi8a2hSSQXCWbsTupd5ljyfkAFpYawPIdQRk6DPBKzBOvZ6Ybh-lMa+kUfolKkBpcqk5XvdyPusiKVXxlGdSJk8SZusnmKpwDDPip1wNdZLoLS-0vxgYMqQABmlyW8ipb1gGAMAAANLjyZWwGkI4hA4UhVZlGANdxKgoAxOhSAABQAaVERWzctq26hyBQGAQFgGE0IQIaIPjpAQHBxa4ORmrAW2lbQTBNCIU2OCwIQ2AQJByDQf5SAYMg4Ajv0w81jBRfujP9jgRZxcaCpRDVgA6COME0VpK8uKpmM9ev3igCoUpz-XMADI2ZEsRYdfNogTCSbBTZcKOc-qQwPhV5iNZN+2wB0K3K7AAAVOg47ARhcBzmAoBYXwsHgHctb2ebmsTsAoC4MAIiifP97t42Y+H3AQU1phOkLMAd74FAF2dVa45zbrAf+8AGDNGIKQGwdAgHf3kOwQsuAbDry3rsUOdtP6nzAIfPAzUjAGn9rgTgkQ8533-nwFBrtsFKzbnQqh7w4CXznpII4WA-akD3BXQM1d0CgNqA3XIuAqgt2ERDKgLASB8HLucRQnUwBSxlrLKkD5FEG2pKogAnDo9RHpW7t07nbbuGA4JG3UM4VAJghAMEAbrPOLUDhtRzm-Ze0hsDElrtQtkYBraBW+ueSuOct47wYDAGA5BcDECHqQBBChpA-xQT4CwVgc6BCgIsMAFZqFOFgDYwh9igHmzzp4qATx5B1zthQwKrtgh5w4X6PkgDSgsHNnwTWmAg4YEkPNHONgWFUAGGqI4Iz7Hi16NIGAuxUi30qbsVBfDMBwQEZHeue4qgUwElSLMFMDH1ykTI-g8i2iaLFto1RcsNFKJUdLPRBzhFtw7grJWZiMoa2CPmBgLC7EhhygAcR0Fcah-8aD0LqA0BAGDh4Wy8UwKOVwTKsjAG3QhLgGBGD5BiKAOdGCz3eGAAAVs7fONxCC4HfqQIZhAL6EBznxGZxx5mYEWb6W+fAEBR3xaQJeSSYnNQZQbQgutd6IKScgjQ7LUmiqBVcDJzA+kaHeLM8Wk9l7bJgIQCZrA4A5yMPYogGgb59Fvk-E6hCa5R0xZYGJ09KAaG4c-EJlwMprKEWASuYMIYAi2bxFy1I9n+seZ6o5sjTm1HOfdS5MtrlaijTs3R+iJGeueSYpW9Tb731FVuZI3ChDBENe4nIV83HIsIUvc2CBpCsHYJwXWxAUCCJwLgdeABNSIEKDWdNNX8agSwcVEO3sXUglKUhZrNjvDJ8dEAIBFQg1A7xOBsA4GAAAIpoJI9Q10ACFl71GYDnR+dTOmYpIXHZddawB8jgJ7ZewQBU5y3E8E+2TH55xri22tN8Ibr3EHbDepte2kmOACQgprRXK3aZgOAGsUgcFgGWh+tTP0VonqgUkMB221L7aBiFZDIiWkLlETp+hXZ4IAGIgcwCoVkni8AUV8I0ZZGA3VWvrt6pYfrnI7KDc5ENldoPkAqE3A2-NM6ekKAAPhVvMAAJIoTOFRmKECpHkAMcEqQenMAAeQAEpgAU0plTyiMoAFYABsWndMGaM8gZTwrlEAHYAAc+jtOtGVoYp5xjXlgEUlywgj8qQ31gWbLBVxdn+ivqKnOlL5DXtvSyjAuxg42DDfwdBK85AKBvdUpWFTWUmwBNQowCBUDwFsBlvgIzBnHFq+8YZgw7Y2H7bV6g4yBBVbgO8XEaCYV4FIN2nJeq7b5ijrIGRGAuCUEHZi4d2TKk9OJOoZqXX9WigUHALgXAHAaGyah1tYAACCrA5DiyoQazhqw3H33fTw5Yc8mzRwS3lnc5BOMYBzhwLBor4saDnjvYIrIsA8WmNvA1cYABamBTY5ySeLd9IZUDgwHdBzphpljxDWxdjA8Avt2yR0ywhw6UsYCMBD+pmggA)

I hope this helps, let me know if I can clarify any of the above! To be clear, all of above modeling considerations are ultimately just DynamoDB modeling considerations. ElectroDB doesn't introduce additional complexity here, just simplifies the implementation of the same patterns you'd use with the simple DocClient.
